src/main/java/pl/edu/mimuw/chess/ChessGame.java

```diff
@@ -0,0 +1,76 @@
+package pl.edu.mimuw.chess;
+
+import pl.edu.mimuw.chess.pieces.KingPiece;
+import pl.edu.mimuw.chess.pieces.Piece;
+
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class ChessGame {
+  public final ChessPlayer whitePlayer;
+  public final ChessPlayer blackPlayer;
+  public final ChessBoard board;
+  public final Random rng;
+  public Piece.Color nextMove;
+
+  public ChessGame(ChessPlayer whitePlayer, ChessPlayer blackPlayer, ChessBoard board, Random rng) {
+    this.whitePlayer = whitePlayer;
+    this.blackPlayer = blackPlayer;
+    this.board = board;
+    this.nextMove = Piece.Color.WHITE;
+    this.rng = rng;
+  }
+
+  public ChessGame(Random rng) {
+    this.board = new ChessBoard();
+    this.whitePlayer = new ChessPlayer(
+      Piece.Color.WHITE,
+      board.getPieces().stream().filter((el) -> el.color == Piece.Color.WHITE).collect(Collectors.toList())
```

Comment:
mam nadzieje że `==` został tu użyty z pełną świadomością, ż można, bo to enum (wyżej było `equals`, które zawsze działa)